catnip_uploader/catnip_uploader.py

```diff
@@ -5,32 +5,125 @@
 import io
 import os
 import time
+import json
+import sys
 
 if platform.system() == "Windows":
     DEFAULT_COMPORT = "COM1"
 else:
```

Comment:
What about macOS? It's not mandatory, but it could be nice to have. Although it could be better to show a ports list to the user, in case default ports are busy by other devices.

Reply:
@DeimosHall can you test if this works on MacOS?

Reply:
@sabas1080 sure